stdlib/public/core/Duration.swift

```diff
@@ -0,0 +1,231 @@
+    let seconds = _attoseconds / 1_000_000_000_000_000_000
+    let attoseconds =
+      Int64((_attoseconds - seconds * 1_000_000_000_000_000_000))
+    return (Int64(seconds), attoseconds)
+  }
+}
+
+@available(SwiftStdlib 9999, *)
+extension Duration {
+  @available(SwiftStdlib 9999, *)
+  public static func seconds<T: BinaryInteger>(_ seconds: T) -> Duration {
+    return Duration(_attoseconds: _Int128(seconds) *
+                                 1_000_000_000_000_000_000)
+  }
+
+  @available(SwiftStdlib 9999, *)
+  public static func seconds(_ seconds: Double) -> Duration {
+    return Duration(_attoseconds: _Int128(seconds *
+                                 1_000_000_000_000_000_000))
+  }
+
+  @available(SwiftStdlib 9999, *)
+  public static func milliseconds<T: BinaryInteger>(
+    _ milliseconds: T
+  ) -> Duration {
+    return Duration(_attoseconds: _Int128(milliseconds) *
+                                 1_000_000_000_000_000)
+  }
+
+  @available(SwiftStdlib 9999, *)
+  public static func milliseconds(_ milliseconds: Double) -> Duration {
+    return Duration(_attoseconds: _Int128(milliseconds *
+                                 1_000_000_000_000_000))
+  }
+
+  @available(SwiftStdlib 9999, *)
+  public static func microseconds<T: BinaryInteger>(
+    _ microseconds: T
+  ) -> Duration {
+    return Duration(_attoseconds: _Int128(microseconds) *
+                                 1_000_000_000_000)
+  }
+
+  @available(SwiftStdlib 9999, *)
+  public static func microseconds(_ microseconds: Double) -> Duration {
+    return Duration(_attoseconds: _Int128(microseconds *
+                                 1_000_000_000_000))
+  }
+
+  @available(SwiftStdlib 9999, *)
+  public static func nanoseconds<T: BinaryInteger>(
+    _ nanoseconds: T
+  ) -> Duration {
+    return Duration(_attoseconds: _Int128(nanoseconds) *
+                                 1_000_000_000)
+  }
+}
+
+@available(SwiftStdlib 9999, *)
+extension Duration: Codable { 
+  private enum CodingKeys: String, CodingKey {
+    case attoseconds = "attoseconds"
+  }
+
+  private enum DecodingFailure: Error, CustomStringConvertible {
+    case conversionFailure(String)
+
+    var description: String {
+      switch self {
+      case .conversionFailure(let str):
+        return "Unable to convert \(str) to Int128"
+      }
+    }
+  }
+  
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    let str = try container.decode(String.self, forKey: .attoseconds)
+    guard let attoseconds = _Int128(str, radix: 10) else {
+      throw DecodingFailure.conversionFailure(str)
+    }
+    self.init(_attoseconds: attoseconds)
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(String(_attoseconds, radix: 10), forKey: .attoseconds)
+  }
+}
+
+@available(SwiftStdlib 9999, *)
+extension Duration: Hashable { }
```

Comment:
Super minor nit: I recommend spelling out the `hash(into:)` implementation here, if only for consistency with `==`. (E.g., if we end up making one of them inlinable at some point, it's good to have the other's definition present so that we don't forget to update it to match.)

Reply:
done